File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    # 정렬
    people.sort()
    people = deque(people)
    cnt = 0
    # 가장 무거운 사람과 가장 가벼운 사람의 합이 <= limit 이면 같이 탈출 아니면 뒤에사람만 혼자 탈출
    while people:
        if len(people) == 1:
            cnt += 1
            break
        if people[0] + people[-1] <= limit:
            people.pop()
            people.popleft()
        else:
            people.pop()
        cnt += 1
    #print(cnt)
    return cnt

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        /*
        가장 무게가 많은사람과 가장 적은 사람의 합이 제한 초과이면
        무게 많은사람은 혼자 타기
        이하이면 같이 타기
        i<j일 때 까지
        */
        int cnt=0;
        int i=0;
        int j=people.length-1;
        Arrays.sort(people);
        while(i<=j){
            int max = people[j];
            int min = people[i];
            if(max+min <= limit){
                i++;
                j--;
            }
            else{
                j--;
            }
            cnt++;
        }
        return cnt;
    }
}